disvae/losses.py

```diff
@@ -167,6 +169,100 @@ def __call__(self, data, recon_data, latent_dist, is_train, storer):
         return loss
 
 
+class FactorKLoss(BaseLoss):
+    """
+        Compute the Factor-VAE loss as per Algorithm 2 of [1]
+
+        Parameters
+        ----------
+        is_color : bool
+            Whether the image are in color.
+
+        discriminator : disvae.discriminator.Discriminator
+
+        optimizer_d : torch.optim.adam.Adam
+
+        device : torch.device
+
+        beta : float, optional
+            Weight of the TC loss term.
+
+        References :
+            [1] Kim, Hyunjik, and Andriy Mnih. "Disentangling by factorising."
+            arXiv preprint arXiv:1802.05983 (2018).
+        """
+
+    def __init__(self, is_color, discriminator, optimizer_d, device, beta=40.):
+        super().__init__(is_color)
+        self.beta = beta
+        self.device = device
+        self.discriminator = discriminator
+        self.optimizer_d = optimizer_d
+
+    def __call__(self, data, model, optimizer, is_train, storer):
+        storer = self._pre_call(is_train, storer)
+
+        # If factor-vae split data into two batches
+        batch_size = data.size(dim=0)
+        half_batch_size = batch_size // 2
+        data = data.split(half_batch_size)
+        data1 = data[0].to(self.device)
```

Comment:
The data should already be on the correct device. 

As a side note, even if it weren't, you would not put it on GPU for each batch. Indeed CPU -> GPU takes time, so better do it once (as done now) than for each batch. 

Note that currently it should nto change anything because i'm pretty sure pytorch checks whether it's already on the correct device before moving it :)

Reply:
Thanks, I'll remove it

Reply:
Yann and I agreed offline that the .to(self.device) should be included here.

Reply:
check the current master branch (due to my last PR) I actually call `.to(self.device)` before the if statement

Reply:
Okay, great. Any splitting the data doesn't affect this?

Reply:
no

Reply:
Fair enough. The only reason I changed it was for the loop which now won't be happening.